### PR TITLE
fix docker image entrypoints

### PR DESCRIPTION
**Pull request by opadron**
Also removes extraneous prompt and ssh handling logic.

This is a subset of the changes originally proposed in #14357.

### CODE REVIEW

File: share/spack/docker/entrypoint.bash [hunk 1]
```diff
@@ -1,45 +1,143 @@
...
+    shift
+fi
 
-            echo -n "Make sure you run docker with the --interactive and --tty"
-            echo -n " options."
-            echo
-        ) >&2
+case "$mode" in
+    "spackenv")
+        # Scenario 1: Run as if the image had no ENTRYPOINT
```
Comment (scottwittenburg) [inline share/spack/docker/entrypoint.bash:26]: Thanks for these descriptions of how the various entrypoints are designed to work.

**Review by scottwittenburg — APPROVED** [review]
Looks good to me, thanks @opadron!

**Comment by scottwittenburg** [comment]
Wish I had remembered this before you hit the button, but I think all the docker build failures I've been getting emails about are caused by missing `python3-setuptools` from the list of `apt` packages, seemingly needed to install `pip install boto3` these days.

**Comment by opadron** [comment]
That's strange because I built each of the four images with these changes locally and haven't had a problem.  And this was on a brand new computer, so there's no chance of anything funny going on with docker caches.

**Comment by scottwittenburg** [comment]
I was unable to build locally myself, with the same error I keep getting in emails from DockerHub.  I'll send you a link via email since not everyone who comes here will be able to follow.

**Comment by scottwittenburg** [comment]
> And this was on a brand new computer, so there's no chance of anything funny going on with docker caches.

Hmm, maybe it has to do with caching.  Does DockerHub do any?  To your knowledge?

**Comment by opadron** [comment]
@scottwittenburg See #17143

**Pull request merged**